docs/federation-changes.md

```diff
@@ -0,0 +1,146 @@
+# Federated Store Changes
+
+This page contains information on the changes to Gaffer's Federated Store. This functionality was introduced in version `2.0.0-alpha-0.4` of Gaffer.
+
+## The Federated Operation
+
+The `FederatedOperationChain` was removed and replaced with a new Operation, the `FederatedOperation`. This was added to improve the control you have over how operations are federated.  
+The Federated Operation has 3 key parameters: `operation`, `graphIds` and `mergeFunction`:
+``` json
+{
+    "class": "uk.gov.gchq.gaffer.federatedstore.operation.FederatedOperation",
+    "operation": {
+        "class": "uk.gov.gchq.gaffer.operation.impl.get.GetAllElements"
+    },
+    "graphIds": [ "graphA", "graphB" ],
+    "mergeFunction": {
+        "class": "uk.gov.gchq.gaffer.federatedstore.util.ConcatenateMergeFunction"
+    }
+}
+```
+
+### Required parameter: operation
+
+This is the Operation you wish to be federated to the subgraphs. This can be a single Operation or an OperationChain. If you use an OperationChain, then the whole chain will be sent together to the subgraphs.  
+
+### Optional parameter: graphIds
+
+This is a list of graph IDs which you want to send the operation to. If you have an OperationChain and wish to send each operation to a different graph, then each operation in the chain will need to be wrapped in a FederatedOperation with the `graphIds` set to the required graph. If instead you want to send the whole OperationChain to the same graphs, then this field can be set on one FederatedOperation where the `operation` is an OperationChain.  
+
+If this parameter is not provided, then the default graph IDs for the store will be used. This is the `storeConfiguredGraphIds` for the store if set, if not set then all graphIds will be used.
+
+### Optional parameter: mergeFunction
+
+The `mergeFunction` parameter is the BiFunction you want to use when merging the results from the subgraphs.  
+
+If this parameter is not provided, then the store's `storeConfiguredMergeFunctions` will be used, which have default values but can be overriden. For example, by default when GetElements is used as the operation inside a FederatedOperation, then `ApplyViewToElementsFunction` will be used to merge the results. Instead, this can be overriden, like in the example to something like `ConcatenateMergeFunction`, or even a custom Function.
```

Comment:
Didn't the default behaviour here change compared with using a `FederatedOperationChain` before? I thought concatenating had been the default and now it is not? If so, this should be documented in more detail, might be considered a breaking change.

Reply:
Yes I was going to doc that more in detail in the full FederatedStore docs but I will add some info on this change as it could be breaking.

Reply:
I added some basic docs on what has changed, roughly how results may differ, and how to get old behaviour. I think more detail on each merge function should be given in the core Federated Store docs refactor and then linked to from here.

Reply:
This is great, exactly the right amount of detail. When the full documentation has been refactored we can add some links to that.